python/ray/train/trainer.py

```diff
@@ -320,12 +325,14 @@ def run(
 
         train_func = construct_train_func(train_func, config)
 
+        dataset_spec = _RayDatasetSpec(dataset_or_dict=dataset)
```

Comment:
At some point we should move / copy the trainer files into `ml/train` right? In preparation for replacing the old `train` module.

Reply:
I thought we decided the end state should be to move ray/ml/train to ray/train right?

But yes agreed, we should definitely cleanup the current ray/train in a future PR.

Reply:
Definitely need the cleanup. I think the easiest way is to copy files over and decouple them, but open to other approaches.